tests/Doctrine/Tests/ORM/Functional/SelfReferencingTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+use function count;
+
+class SelfReferencingTest extends OrmFunctionalTestCase
```

Comment:
Please use `GH7877` as the prefix for the classes (as we have on the newer test cases). Also, add `@group GH7877` to the class docblock to be able to easily execute all tests related to that issues.